main.go

```diff
@@ -453,13 +689,92 @@ func getStats(conn net.Conn) (map[string]string, error) {
 	return m, nil
 }
 
+// Get detailed per-server stats from mcrouter using a basic TCP connection
+func getServerStats(conn net.Conn) (map[string]map[string]string, error) {
+	m := make(map[string]map[string]string)
+	fmt.Fprintf(conn, "stats servers\r\n")
+	reader := bufio.NewReader(conn)
+
+	// Iterate over the lines and extract the metric name and value(s)
+	// example lines:
+	// 	 STAT 10.64.16.110:11211:ascii:plain:notcompressed-1000 avg_latency_us:302.991
+	//        pending_reqs:0 inflight_reqs:0 avg_retrans_ratio:0 max_retrans_ratio:0
+	//        min_retrans_ratio:0 up:5; deleted:4875 touched:33069 found:112675373
+	//        notfound:3493823 notstored:149776 stored:3250883 exists:2653 remote_error:32
+	//	 END
+	// In the same line there are two type of info:
+	// * per-server stats about latency, requests, etc.. up to the ';' - (ProxyDestinationBase states)
+	// * per-server breakdown of the memcached responses (STORED, DELETED, etc..) - (Carbon results)
+	//   The memcached responses are listed in carbon_result.thrift, and they are returned/appended only
+	//   when > 0. Special logic needs to be implemented then to set the ones not diplayed to zero,
+	//   since Prometheus doesn't really like metrics appearing/disappearing.
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			return nil, err
+		}
+
+		if line == "END\r\n" {
+			break
+		}
+
+		line_sanitized := strings.TrimRight(line, "\r\n")
+
+		// Split the line in 2 macro-components
+		result := strings.SplitN(line_sanitized, ";", 2)
+
+		server_metrics := strings.Split(strings.Trim(result[0], " "), " ")
+		memcached_responses_metrics := strings.Split(strings.Trim(result[1], " "), " ")
```

Comment:
What I could do in here is something like the following:
1) split by ';'
2) check the len of the result - if not 2, then it means an old version of mcrouter is being used. If so, don't attempt to set/retrieve and memcached per-server metrics.
3) otherwise, populate all the metrics.

The use case 2) should be something that the user might not be aware, so a warning in the logs could be useful but probably annoying after a while. Maybe adding a specific note to README.md could suffice?